src/events/service/eventService.ts

```diff
@@ -3,48 +3,177 @@ import { CreateEventDto } from "../dto/eventCreateDto";
+    const currentPivot = await pivotRepository.findOne({
+      order: { id: "DESC" },
+      where: { eventId: Id },
+    });
+    const { eventId, criteriaId, subCriteriaId, userId } = currentPivot
+      ? currentPivot
+      : { eventId: 0, criteriaId: 0, subCriteriaId: 0, userId: 0 };
+    (await pivotRepository.createQueryBuilder().where({ eventId: Id }).getOne())
+      ? (await pivotRepository
+          .createQueryBuilder()
+          .where({ eventId: Id })
+          .andWhere({ ratingId: idRef.id })
+          .getOne())
+        ? logger.info("data already exists")
+        : await pivotRepository
+            .createQueryBuilder()
+            .insert()
+            .values({
+              ratingId: idRef.id,
+              eventId: eventId,
+              criteriaId: criteriaId,
+              subCriteriaId: subCriteriaId,
+              userId: userId,
+            })
+            .execute()
+      : await pivotRepository.save({
+          eventId: Id,
+          ratingId: idRef.id,
+          criteriaId,
+          subCriteriaId,
+          userId,
+        });
+  }
 
-  async addCriteria(eventId: number, criteriaRef: ICriteriaRef) {
-    const criteria = await this.criteriaRepository.findOne(criteriaRef.id)
-    const event = await this.eventsRepository.findOneById(eventId)
-    event.criteria.push(criteria)
-    return this.eventsRepository.addUsers(event)
+  async addCriteria(Id: number, idRef: IUserRef) {
+    const pivotRepository = await getRepository(Pivot);
+    const currentPivot = await pivotRepository.findOne({
+      order: { id: "DESC" },
+      where: { eventId: Id },
+    });
+    const { eventId, ratingId, subCriteriaId, userId } = currentPivot
+      ? currentPivot
+      : { eventId: 0, ratingId: 0, subCriteriaId: 0, userId: 0 };
+    (await pivotRepository.createQueryBuilder().where({ eventId: Id }).getOne())
+      ? (await pivotRepository
+          .createQueryBuilder()
+          .where({ eventId: Id })
+          .andWhere({ criteriaId: idRef.id })
+          .getOne())
+        ? logger.info("data already exists")
+        : await pivotRepository
+            .createQueryBuilder()
+            .insert()
+            .values({
+              criteriaId: idRef.id,
+              eventId: eventId,
+              ratingId: ratingId,
+              subCriteriaId: subCriteriaId,
+              userId: userId,
+            })
+            .execute()
+      : await pivotRepository.save({
+          eventId: Id,
+          criteriaId: idRef.id,
+          ratingId,
+          subCriteriaId,
+          userId,
+        });
   }
 
-  async addUsers(eventId: number, userRef: IUserRef) {
-    const user = await this.userRepository.findOne(userRef.id)
-    const event = await this.eventsRepository.findOneById(eventId)
-    event.users.push(user)
+  async addSubCriteria(Id: number, idRef: ISubCriteriaRef) {
+    const pivotRepository = await getRepository(Pivot);
```

Comment:
Please move as much as possible into repsotory, and use only our created repositories instead of getRepository